### PR TITLE
Add Kubeflow Components Overview Diagram

### PR DESCRIPTION
**Pull request by andreyvelich**
This diagram shows high-level diagram for Kubeflow components, so users can see which part of MLOps lifecycle Kubeflow covers today.
We can edit this diagram in the future by simply import it to: https://www.drawio.com/

We showed this diagram in the latest Kubeflow Summit: https://youtu.be/HBBJYa4JTNY?list=PL2gwy7BdKoGdrkYIWGeAdKi9ntfxq8FYt

Related: https://github.com/kubeflow/website/issues/3536

/hold for review

### CODE REVIEW

**Comment by andreyvelich** [comment]
cc for the review on the Kubeflow high-level architecture.
@kubeflow/wg-training-leads @kubeflow/wg-pipeline-leads @kubeflow/wg-notebooks-leads 
@jbottum @thesuperzapper @akgraner @vikas-saxena02

File: content/en/docs/started/images/kubeflow-overview.drawio.png
Comment (terrytangyuan) [inline content/en/docs/started/images/kubeflow-overview.drawio.png:1]: Should we store the drawio source file as well so it's easier to modify later

Comment (thesuperzapper) [reply content/en/docs/started/images/kubeflow-overview.drawio.png:1]: @andreyvelich Also, I dont know why/how but this image is somehow interactively linked to draw.io: https://deploy-preview-3650--competent-brattain-de2d6d.netlify.app/docs/started/architecture/

I recommend we raster it down to an SVG to ensure compatibility.

Comment (thesuperzapper) [reply content/en/docs/started/images/kubeflow-overview.drawio.png:1]: Also, your diagram should make it clear that any Kubernetes cluster will work, so perhaps we should replace the minikube logo with the Kubernetes one?

Comment (thesuperzapper) [reply content/en/docs/started/images/kubeflow-overview.drawio.png:1]: Also, I think the components should be ordered by popularity, or possibly the order in which you use them.

For example:

- Top-Left: Kubeflow Pipelines
- Top-Right: Kubeflow Notebooks
- Mid-Left: Central Dashboard
- Mid-Right:  Katib (AutoML)
- Bottom-Left: Training Operator
- Bottom-Right: KServe (Serving)

(You also missed central dashboard, which is a pretty important component).

**Comment by thesuperzapper** [comment]
@andreyvelich I thought your diagram going to be used on the "introduction" page rather than the "architecture"?

And that we would use a diagram like this one on the "architecture" page:

Screenshot 2023-12-15 at 10 22 57

File: content/en/docs/started/images/kubeflow-overview.drawio.png
Comment (andreyvelich) [reply content/en/docs/started/images/kubeflow-overview.drawio.png:1]: >Should we store the drawio source file as well so it's easier to modify later

@terrytangyuan If you upload this image to drawio, you can modify it

Comment (thesuperzapper) [reply content/en/docs/started/images/kubeflow-overview.drawio.png:1]: @andreyvelich I think we should probably disable the "edit" button as I was saying in https://github.com/kubeflow/website/pull/3650#discussion_r1428333592, it seems that [docsy has native draw.io PNG intergration](https://www.docsy.dev/docs/adding-content/diagrams-and-formulae/#diagrams-with-diagramsnet), so to prevent that strange interactivity, we should render it to an SVG.

Comment (andreyvelich) [reply content/en/docs/started/images/kubeflow-overview.drawio.png:1]: > I recommend we raster it down to an SVG to ensure compatibility.

@thesuperzapper For some reason GitHub and Netlify can't properly render the `svg` format from DrawIO: 
https://github.com/kubeflow/website/blob/49cf19175d6a0bd05f2f880130203c4ce49fcdea/content/en/docs/images/kubeflow-overview.drawio.svg
Any thoughts ?

Comment (andreyvelich) [reply content/en/docs/started/images/kubeflow-overview.drawio.png:1]: >Also, your diagram should make it clear that any Kubernetes cluster will work, so perhaps we should replace the minikube logo with the Kubernetes one?

I think, this indicates various Kubernetes distributions where you can run Kubeflow and SDK, Web UI, and `kubectl` is an interfaces how user can interact with Kubeflow and Kubernetes APIs.

If you think that we should indicate Kubernetes, I can draw the Kubernetes at the bottom, how do you like it?
kubeflow-overview drawio

Comment (andreyvelich) [reply content/en/docs/started/images/kubeflow-overview.drawio.png:1]: >Also, I think the components should be ordered by popularity,

@thesuperzapper How are you going to track the components popularity ? I tried to order them by MLOps requirements. On the left side users can see components for Train -> Tune -> Serve their ML workloads, on the right side they can see components for additional ML activities (e.g. develop ML in Notebooks, build ML Pipelines, work with ML metadata).

What do others think about it @kubeflow/wg-training-leads @kubeflow/wg-pipeline-leads @jbottum @kubeflow/wg-notebooks-leads ?

Comment (andreyvelich) [reply content/en/docs/started/images/kubeflow-overview.drawio.png:1]: > (You also missed central dashboard, which is a pretty important component).

Is central dashboard more important than ML Metadata for MLOps story ? Users can always find the all Kubeflow stack in your Kubeflow Stack diagram which explain all Kubeflow components.

**Comment by andreyvelich** [comment]
> I thought your diagram going to be used on the "introduction" page rather than the "architecture"?

Yes, let me move it to introduction for now. I hope, we have time to discuss it during tomorrow's community call to hear the feedback from others.

File: content/en/docs/started/introduction.md [hunk 1]
```diff
@@ -10,6 +10,13 @@ recreate other services, but to provide a straightforward way to deploy
 best-of-breed open-source systems for ML to diverse infrastructures. Anywhere
 you are running Kubernetes, you should be able to run Kubeflow.
 
+The following diagrams shows the main Kubeflow components to cover each step of ML lifecycle
```
Comment (hbelmiro) [inline content/en/docs/started/introduction.md:13]: Small typo.

```suggestion
The following diagram shows the main Kubeflow components to cover each step of ML lifecycle
```

**Comment by vikas-saxena02** [comment]
@andreyvelich , the diagram looks great to me!!
I just have one suggestion though, for the ML Libraries section on top. Is it possible to provide names for the ones which do not have it mentioned in the icon. For non-technical users (managers, BAs etc) it will be confusing as they do not know which library is that.

File: content/en/docs/started/images/kubeflow-overview.drawio.png
Comment (jbottum) [reply content/en/docs/started/images/kubeflow-overview.drawio.png:1]: isn't the Web UI the Central Dashboard ?

Comment (thesuperzapper) [reply content/en/docs/started/images/kubeflow-overview.drawio.png:1]: @jbottum @andreyvelich perhaps we should re-word the "Web UI" to "Web Dashboard"?

Comment (thesuperzapper) [reply content/en/docs/started/images/kubeflow-overview.drawio.png:1]: @andreyvelich I think we can swap the "layers" to more logically represent what runs "on top" of what.

My idea for ordering (from top to bottom):

1. __Libraries__ (Jupyter / Tensorflow / Torch / ...)
     - Yellow Horizontal Bar
3. __Interfaces__ (SDKs / Web Dashboard / Kubectl)
     - Blue Arrow
5. __Kubeflow__
     - Blue Arrow
7. __Kubernetes__
     - Yellow Horizontal Bar
9. __Platforms__ (AWS / GCP / On-Prem / ...)

Some other notes:

- Minikube is not really a platform, so we can remove it (or possibly replace it with "Local Deployment", with a picture of a laptop).
- We should put a dotted line around the "Platforms" group, like the "Libraries" one.

Comment (andreyvelich) [reply content/en/docs/started/images/kubeflow-overview.drawio.png:1]: Web UI is the combination of all UI that Kubeflow offers today (e.g. Central Dashboard, Katib, Pipelines, Volumes, Tensorboards, etc.)

**Comment by andreyvelich** [comment]
> @andreyvelich , the diagram looks great to me!! I just have one suggestion though, for the ML Libraries section on top. Is it possible to provide names for the ones which do not have it mentioned in the icon. For non-technical users (managers, BAs etc) it will be confusing as they do not know which library is that.

That's good point @vikas-saxena02. I will make this change to the diagram.

File: content/en/docs/started/images/kubeflow-overview.drawio.png
Comment (andreyvelich) [reply content/en/docs/started/images/kubeflow-overview.drawio.png:1]: @thesuperzapper I agree that Kubernetes should go above the platforms and I can change minikube to local deployment, but I am not sure if SDK/Web UI/kubectl should be between Kubeflow and libraries.

My thoughts are that interfaces (SDK, Web UI, and kubectl) is a layer between Kubeflow components and Kubernetes. For example, when user run `create_job` Training Operator SDK API, it creates resources in Kubernetes clusters.
Or when user clicks create Experiment button on the Katib UI it creates custom resources in Kubernetes.

Any thoughts @thesuperzapper @jbottum @johnugeorge @vikas-saxena02 ?

**Comment by andreyvelich** [comment]
@thesuperzapper @terrytangyuan @johnugeorge @tenzen-y @hbelmiro @vikas-saxena02 @jbottum Any other suggestions before we can merge this initial diagram ?

**Review by hbelmiro — APPROVED** [review]

**Comment by thesuperzapper** [comment]
@andreyvelich I like where you are going but I think it can still be made more clear.

I have taken a stab at it based on your design, I went back and forward on keeping the "Web UI" box, but settled on replacing "ML Metadata" with "Central Dashboard" because that's similar to what we do on the kubeflow.org homepage.

The following file is in SVG draw.io format if you want to modify it further (right click to download as SVG):

kubeflow-intro-diagram drawio

PS: we should use SVG rather than PNG for the website (don't forget to click "embed images" when exporting from draw.io)

**Review by terrytangyuan — APPROVED** [review]
Great work! We can always improve/iterate later.

/lgtm

**Comment by thesuperzapper** [comment]
@andreyvelich based on your comment I think we can add the "Model Training" / etc labels underneath each component to help new users understand what each box means.

I also think that it makes sense to put the development tools on the left side (the same order we use on the homepage).

For example:
kubeflow-intro-diagram-2 drawio

And the compressed file version:
[kubeflow-intro-diagram-2.drawio.svg.zip](https://github.com/kubeflow/website/files/13876031/kubeflow-intro-diagram-2.drawio.svg.zip)

**Comment by thesuperzapper** [comment]
@andreyvelich Based on a few comments I have made the labels a lot less prominent 

kubeflow-intro-diagram-3 drawio

[kubeflow-intro-diagram-3.drawio.svg.zip](https://github.com/kubeflow/website/files/13876474/kubeflow-intro-diagram-3.drawio.svg.zip)

**Comment by andreyvelich** [comment]
I made the final changes based on the discussion.
Please provide your final comments and we can merge it.
/assign @thesuperzapper @terrytangyuan @johnugeorge @tenzen-y @hbelmiro @vikas-saxena02 @jbottum @james-jwu @zijianjoy

**Review by hbelmiro — APPROVED** [review]

**Comment by zijianjoy** [comment]
/lgtm
/approve

Thank you @andreyvelich and everyone for your help on this PR!

**Comment by tenzen-y** [comment]
@andreyvelich Thank you for this improvement! This looks great :)
